.github/workflows/release-gh-publish.yaml

```diff
@@ -0,0 +1,88 @@
+          git config --global user.name github-actions[bot]
+          git checkout odh-${{ env.VERSION }}
+          git tag -a -m v${{ env.VERSION }} v${{ env.VERSION }}
+          git push origin v${{ env.VERSION }}
+      - name: Get component release notes
+        id: release-notes
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const script = require('./.github/scripts/get-component-release-notes.js')
+            script({github, core})
+      - name: Create GH release
+        uses: softprops/action-gh-release@v2
+        with:
+          body: ${{ steps.release-notes.outputs.release-notes-body }}
+          tag_name: v${{ env.VERSION }}
+          generate_release_notes: true
+          append_body: true
+          make_latest: true
+  create-community-operators-pr:
+    needs: [gh-release]
+    name: Create community operators prod pr # https://github.com/redhat-openshift-ecosystem/community-operators-prod
+    runs-on: ubuntu-latest
+    steps:
+      - uses: tibdex/github-app-token@v1
+        id: generate-token
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+      - name: Set env variables
+        uses: opendatahub-io/opendatahub-operator/.github/actions/set-shared-env@incubation
+        with:
+          pat-token: ${{ steps.generate-token.outputs.token }}
+      - name: Checkout opendatahub-operator
+        uses: actions/checkout@v4
+        with:
+          path: ./opendatahub-operator
+          ref: v${{ env.VERSION }}
+      - name: Checkout redhat-openshift-ecosystem/community-operators-prod
+        uses: actions/checkout@v4
+        with:
+          token: <PAT> # We need a token with repo rights
+          repository: redhat-openshift-ecosystem/community-operators-prod # replaced with redhat-openshift-ecosystem/community-operators-prod
+          path: ./community-operators-prod
+      - name: Copy bundle files and add annotation
+        run : |
+          ls -la
+          # cd community-operators-prod
+          # mkdir -p community-operators-prod/operators/opendatahub-operator/${{ env.VERSION }}
+          # ls -la
+          # cp -a ../opendatahub-operator/bundle/. operators/opendatahub-operator/${{ env.VERSION }}/
+          # echo "
+          #   # OpenShift specific version
+          #   com.redhat.openshift.versions: v4.9" >> operators/opendatahub-operator/${{ env.VERSION }}/metadata/annotations.yaml
+          # sed -i -e "s|image: REPLACE_IMAGE:latest.*|image: quay.io/opendatahub/opendatahub-operator:v${{ env.VERSION }}|g"  operators/opendatahub-operator/${{ env.VERSION }}/manifests/opendatahub-operator.clusterserviceversion.yaml
+          # git status
+      - name: Create community operators prod pr
+        uses: peter-evans/create-pull-request@v6
+        with:
+          path: ./community-operators-prod
+          token: <PAT> # We need a token with repo rights
```

Comment:
Here, for some reason the token generated by github app does not work. I created an issue in the library: https://github.com/peter-evans/create-pull-request/issues/2848 .
Also I created a shell script to mimic the pr creation process, but unfortunately even that flow works with a PAT and not with a token generated from GH app. Any suggestions/help here is much appreciated.

Reply:
the PAT to be used needs the `repo` scope.